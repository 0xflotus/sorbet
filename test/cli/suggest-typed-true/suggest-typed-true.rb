def true_func
end
