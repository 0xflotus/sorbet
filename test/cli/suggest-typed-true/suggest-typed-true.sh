#!/bin/bash

tmp="$(mktemp -d)"
cwd="$(pwd)"
cd "$tmp" || exit 1
separator() {
    echo -------------------------
}

cp "$cwd"/test/cli/suggest-typed-true/*.rb "$tmp/"

"$cwd/main/sorbet" --silence-dev-message --suggest-typed does-not-exist.rb 2>&1
separator

"$cwd/main/sorbet" --silence-dev-message --suggest-typed --typed=ruby does-not-exist.rb 2>&1
separator

"$cwd/main/sorbet" --silence-dev-message --suggest-typed --typed=true does-not-exist.rb 2>&1
separator

"$cwd/main/sorbet" --silence-dev-message -a --suggest-typed --typed=true does-not-exist.rb 2>&1
separator

"$cwd/main/sorbet" --silence-dev-message --suggest-typed --typed=true suggest-typed-true.rb 2>&1
separator

"$cwd/main/sorbet" --silence-dev-message -a --suggest-typed --typed=true suggest-typed-true.rb 2>&1
cat suggest-typed-true.rb
separator

cp "$cwd/test/cli/suggest-typed-true/suggest-typed-true.rb" "$tmp"
"$cwd/main/sorbet" --silence-dev-message --suggest-typed --typed=true suggest-typed-true.rb suggest-typed-with-error.rb 2>&1
separator

"$cwd/main/sorbet" --silence-dev-message --suggest-typed --typed=true empty.rb 2>&1
separator

rm -r "$tmp"
