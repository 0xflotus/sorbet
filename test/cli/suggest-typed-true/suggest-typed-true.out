--suggest-typed requires --typed=true or higher.
-------------------------
--suggest-typed requires --typed=true or higher.
-------------------------
does-not-exist.rb: File Not Found http://go/e/1001
Errors: 1
-------------------------
does-not-exist.rb: File Not Found http://go/e/1001
Errors: 1
-------------------------
suggest-typed-true.rb:1: You could add `# typed: true` http://go/e/7022
     1 |1
        ^
  Autocorrect: Use `-a` to autocorrect
    suggest-typed-true.rb:1: Insert `# typed: true
`
     1 |1
        ^
Errors: 1
-------------------------
suggest-typed-true.rb:1: You could add `# typed: true` http://go/e/7022
     1 |1
        ^
  Autocorrect: Done
    suggest-typed-true.rb:1: Inserted `# typed: true
`
     1 |1
        ^
Errors: 1
# typed: true
1
-------------------------
suggest-typed-with-error.rb:1: Method `foo` does not exist on `T.class_of(<root>)` http://go/e/7003
     1 |foo
        ^^^

suggest-typed-true.rb:1: You could add `# typed: true` http://go/e/7022
     1 |1
        ^
  Autocorrect: Use `-a` to autocorrect
    suggest-typed-true.rb:1: Insert `# typed: true
`
     1 |1
        ^
Errors: 2
-------------------------
empty.rb:1: You could add `# typed: true` http://go/e/7022
     1 |
       ^
  Autocorrect: Use `-a` to autocorrect
    empty.rb:1: Insert `# typed: true
`
     1 |
       ^
Errors: 1
-------------------------
